### PR TITLE
PSP-722 Disable Agency dropdown for MOTI tenant

### PR DESCRIPTION
**Pull request by asanchezr**
## CITZ
image

## MOTI
image

### CODE REVIEW

File: frontend/src/utils/YupSchema.ts [hunk 1]
```diff
@@ -14,9 +14,13 @@ Yup.addMethod(Yup.string, 'optional', function optional() {
...
-  agency: Yup.number()
-    .min(1, 'Invalid Agency')
-    .required('Required'),
+  showAgency: Yup.boolean(),
+  agency: Yup.number().when('showAgency', {
+    is: true,
+    then: Yup.number()
+      .min(1, 'Invalid Agency')
+      .required('Required'),
+  }),
```
Comment (asanchezr) [inline frontend/src/utils/YupSchema.ts:23]: Had to introduce a `showAgency` form field (hidden from the UI) to control whether `agency` is a required field. 

* MOTI: no agency (optional)
* CITZL agency is required

File: frontend/src/features/admin/access-request/AccessRequestPage.MOTI.test.tsx [hunk 2]
```diff
@@ -132,52 +149,64 @@ describe('AccessRequestPage functionality', () => {
...
+  it('displays enabled roles', () => {
+    const { queryByText } = testRender();
+    expect(queryByText('roleVal')).toBeVisible();
+  });
 
-    it('Does not display private roles', () => {
-      const { queryByText } = testRender();
-      expect(queryByText('privateRole')).toBeNull();
-    });
+  it('does not display disabled roles', () => {
```
Comment (devinleighsmith) [inline frontend/src/features/admin/access-request/AccessRequestPage.MOTI.test.tsx:180]: is there a test for the agency not being visible? I suppose technically the snapshot would cover it. Might be better to make it explicit. What do you think?

Comment (devinleighsmith) [reply frontend/src/features/admin/access-request/AccessRequestPage.MOTI.test.tsx:180]: Kind of covered by the showAgency functionality, up to you.

**Review by devinleighsmith — APPROVED** [review]

File: frontend/src/features/admin/access-request/AccessRequestPage.MOTI.test.tsx (hunk 2 above)
Comment (asanchezr) [reply frontend/src/features/admin/access-request/AccessRequestPage.MOTI.test.tsx:180]: I'm okay adding a text to explicitly check for non-existence. My thoughts were that the snapshot covers that but you are right that snapshot could be updated and we would loose that check

**Pull request merged**